File: 1765-map-of-highest-peak/1765-map-of-highest-peak.java
class Solution {
    public int[][] highestPeak(int[][] isWater) {
        int[][] dirs = {{0, 1}, {0, -1}, {-1, 0}, {1, 0}};
        int m = isWater.length, n = isWater[0].length;
        int[][] height = new int[m][n];
        for (int[] row : height) Arrays.fill(row, -1);
        Queue<int[]> q = new LinkedList<>();
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (isWater[i][j] == 1) {
                    height[i][j] = 0;
                    q.offer(new int[]{i, j});
                }
            }
        }
        while (!q.isEmpty()) {
            int[] curr = q.poll();
            int i = curr[0], j = curr[1];
            for (int[] dir : dirs) {
                int x = i + dir[0], y = j + dir[1];
                if (x >= 0 && y >= 0 && x < m && y < n && height[x][y] == -1) {
                    height[x][y] = height[i][j] + 1;
                    q.offer(new int[]{x, y});
                }
            }
        }
        return height;
    }
}

File: 1765-map-of-highest-peak/1765-map-of-highest-peak.py
class Solution:
    def highestPeak(self, isWater: List[List[int]]) -> List[List[int]]:
        dirs = [(0, 1), (0, -1), (-1, 0), (1, 0)]
        m, n = len(isWater), len(isWater[0])
        height = [[-1] * n for _ in range(m)]
        q = deque()
        for i in range(m):
            for j in range(n):
                if isWater[i][j] == 1:
                    height[i][j] = 0
                    q.append((i, j))
        while q:
            i, j = q.popleft()
            for dx, dy in dirs:
                x, y = i + dx, j + dy
                if 0 <= x < m and 0 <= y < n and height[x][y] == -1:
                    height[x][y] = height[i][j] + 1
                    q.append((x, y))
        return height